### PR TITLE
Add ADR for moving to React testing library, and migrate several test files from Enzyme

### PR DESCRIPTION
**Pull request by azangru**
## Description
An ADR containing rationale for moving from `enzyme` to `react-testing-library`, and some test files migrated over to react-testing-library to serve as examples.

### CODE REVIEW

**Review by ens-ap5 — APPROVED** [review]
I can't speak for the unit tests, not being familiar with RTL, but the ADR is very thorough and convincing!

File: src/ensembl/doc/architecture/decisions/0016-react-testing-library.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Over a year and a half later, it has become clear that the extensive and elegant API of `enzyme`, with its multiple convenience functions for selecting and inspecting components, comes at a significant cost. In order to provide this API, `enzyme` relies on the knowledge of React's internals, which can, and do, change between releases. Every time this happens, `enzyme` has to catch up, and projects relying on `enzyme` need to wait until `enzyme` becomes capable of supporting the latest version of React before being able to update to this version.
```
Comment (imransl) [inline src/ensembl/doc/architecture/decisions/0016-react-testing-library.md:15]: > In order to provide this API, `enzyme` relies on the knowledge of React's internals, which can, and do, change between releases.

This made me wonder if we have other dependencies that are similar to `enzyme` which we need to be concerned of.

**Review by imransl — APPROVED** [review]

File: src/ensembl/doc/architecture/decisions/0016-react-testing-library.md (hunk 1 above)
Comment (azangru) [reply src/ensembl/doc/architecture/decisions/0016-react-testing-library.md:15]: None that I'm aware of.

The worst offender now is `storybook`, which went so wild with the way it handles its builds that it's incompatible with webpack 5 at the moment. But that's a different story.

File: src/ensembl/doc/architecture/decisions/0004-unit-tests.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # 4. Unit Tests
 
-Date: 2018-08-30
+Date: 2018-08-30; Last update date: 2019-04-09
 
 ## Status
```
Comment (ens-st3) [inline src/ensembl/doc/architecture/decisions/0004-unit-tests.md:6]: We should link to 014 so that it's clear we're not using Enzyme

Comment (azangru) [reply src/ensembl/doc/architecture/decisions/0004-unit-tests.md:6]: @ens-st3, added the link in https://github.com/Ensembl/ensembl-client/pull/407/commits/81d63e87641be9fc44578ed58100a1f85dbf05bd

**Review by msmanoj — APPROVED** [review]

**Review by ens-st3 — APPROVED** [review]

**Pull request merged**